datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -875,14 +914,14 @@ macro_rules! get_data_page_statistics {
                 Some(DataType::Date32) => Ok(Arc::new(Date32Array::from_iter([<$stat_type_prefix Int32DataPageStatsIterator>]::new($iterator).flatten()))),
                 Some(DataType::Date64) => Ok(
                     Arc::new(
-                        Date64Array::from([<$stat_type_prefix Int32DataPageStatsIterator>]::new($iterator)
+                        Date64Array::from_iter([<$stat_type_prefix Int32DataPageStatsIterator>]::new($iterator)
                             .map(|x| {
                                 x.into_iter()
-                                .filter_map(|x| {
+                                .map(|x| {
```

Comment:
I think these changes are similar to what is in https://github.com/apache/datafusion/pull/11295 -- perhaps we should merge that one first (as it has test) and then merge this PR

Reply:
I removed the unnecessary collect, so it should be good to go once the tests have passed.

I suspect the remaining cases where we are using collect could be made more efficient using the Builder pattern?